### PR TITLE
Add Travis MacOS build, GitHub Actions MacOS/Windows build and adjust…

### PR DESCRIPTION
**Pull request by asfernandes**
… MacOS build to be (almost) relocatable.

While MacOS people seems to be ok with fixed locations for applications and libraries, this complicates a lot the (post)
build process, needing to change each id and rpaths in a very error prone process.

Relocatable binaries makes this a lot easier, but unfortunately "restricted" (chmod +s, like firebird executable)
programs cannot use @loader_path or @executable_path in its rpath.

So the solution has to make internal libraries relocatable and make rpath of firebird fixed. Also, as the ecosystem
seems to use fixed path, the id of fbclient.dylib has set to its fixed path.

Also MacOS post build makefile has adjusted to allow creation of packages for the debug build.

The MacOS build could still be improved with some scripts to build ICU (instead of done directly in the CI scripts,
but I leave that for now) and copies its files to our lib path. However situation seems to be better than before in
relation to ICU and TomMath.

Note: Linux build is not working in GitHub Actions. It segfaults when running (exiting) utilities.
I had this problem lot's of time in the past, maybe it's not completely fixed in v3.

### CODE REVIEW

**Comment by AlexPeshkoff** [comment]
I doubt - is there real need to have S bit set on Mac for firebird?

**Comment by asfernandes** [comment]
> I doubt - is there real need to have S bit set on Mac for firebird?

It's also used in Linux, no?

File: builds/install/arch-specific/darwin/Makefile.in [hunk 1]
```diff
@@ -29,48 +31,48 @@ package_firebird:
 			-e 's/_MINOR_VERS_/$(FB_MINOR_VERS)/g' \
 			-e 's/_REV_NO_/$(FB_REV_NO)/g' \
 			../builds/install/arch-specific/darwin/Info.plist \
-			> $(ROOT)/gen/Release/frameworks/Firebird3.framework/Resources/Info.plist
+			> $(ROOT)/gen/$(TARGET)/frameworks/Firebird3.framework/Resources/Info.plist
```
[the remark this answers is not in the document]
Comment (asfernandes) [reply builds/install/arch-specific/darwin/Makefile.in]: It's only about flexibility, in fact I don't ever tested it. But may be useful when distributing private build for test.

**Comment by AlexPeshkoff** [comment]
No. S-bit is not used on linux since 2.5. We had a root-owned fb_lock_mgr process required to deliver signals between processes that could not do it themselves due to not enough rights. After switching from unix signals to blocking thread that binary became not needed at all.

And S-bit was never (as long as my memory works) set on linux for fb_inet_server of fbserver.

File: builds/install/arch-specific/darwin/Makefile.in (hunk 1 above)
[the remark this answers is not in the document]
Comment (AlexPeshkoff) [reply builds/install/arch-specific/darwin/Makefile.in]: If it appears useful I see no problems

**Comment by asfernandes** [comment]
> No. S-bit is not used on linux since 2.5. We had a root-owned set-owner-rights fb_lock_mgr binary required to deliver signals between processes that could not do it themselves due to not enough rights. After switching from unix signals to blocking thread that binary became not needed at all.
> 
> And S-bit was never (as long as my memory works) set on linux for fb_inet_server of fbserver.

Then I think it should not be set as well. In my test it worked and it would not be necessary to set rpath of firebird executable to fixed path.

BTW there is duplicated MacOS file builds/install/arch-specific/darwin/install-script in builds/posix/install-script.

**Comment by paulbeach** [comment]
> BTW there is duplicated MacOS file builds/install/arch-specific/darwin/install-script in builds/posix/install-script.

Indeed, and it needs removing. The install-script in arch-specific/darwin is the one used by Makefile.install

**Pull request merged**